src/main/java/seedu/address/model/person/Partner.java

```diff
@@ -86,6 +86,11 @@ public Role getRole() {
         return Role.PARTNER;
     }
 
+    @Override
+    public String getRoleAsLowerCaseString() {
+        return Role.PARTNER.toLowerCaseString();
```

Comment:
Doing this instead might still be violating LOD? Since you are accessing an attribute of Role before calling its method.

Reply:
Yeah it still violates. I think having a role field would be a better idea.